gen3sdk/R/Gen3AuthHelper.R

```diff
@@ -0,0 +1,63 @@
+library(httr)
+library(jsonlite)
+
+Gen3AuthHelper <- setRefClass("Gen3AuthHelper",
+
+# Gen3 auth helper class for use with requests auth.
+
+# Implements requests.auth.AuthBase in order to support JWT authentication.
+# Generates access tokens from the provided refresh token file or string.
+# Automatically refreshes access tokens when they expire.
+
+# Args:
+#   endpoint (str): The URL of the data commons.
+#   refresh_file (str): The file containing the downloaded json web token.
+
+#   Examples:
+#       This generates the Gen3Auth class pointed at the sandbox commons while
+#       using the credentials.json downloaded from the commons profile page.
+
+#       >>> auth <- Gen3AuthHelper("https://nci-crdc-demo.datacommons.io", refresh_file="credentials.json")
+
+    fields = list(
+        endpoint= "character",
+        refresh_file= "character"
+    ),
+
+    methods = list(
+        initialize = function(endpoint="", refresh_file="") {
+            .self$endpoint <- endpoint
+            .self$refresh_file <- refresh_file
+        },
+        
+        get_access_token = function() {
+            if (.self$endpoint=="") {
```

Comment:
these checks should go in initialize

Reply:
If I don't supply endpoint and refresh_file with default values, it makes these checks obsolete because the initialize method will enforce missing values. However, since the other classes (Submission, File, Indexd) have AuthHelper as an instance variable, endpoint and refresh_file must be supplied with default values. Otherwise the R package will not even install. I can't figure out a way to put theses checks in initialize.